tests/test_telemetry.py

```diff
@@ -359,3 +359,42 @@ def test_app_dependencies_loaded(self):
         for dependency, seen in seen_loaded_dependencies.items():
             if not seen:
                 raise Exception(dependency + " not recieved in app-dependencies-loaded message")
+
+    def test_app_started_product_info(self):
+        """Assert that product information is accurately reported by telemetry"""
+
+        def validator(data):
+            if data["request"]["content"].get("request_type") == "app-started":
+                content = data["request"]["content"]
+                if content.get("request_type") == "app-product-change":
+                    products = content["payload"]["products"]
+                    try:
+                        products["appsec"]
```

Comment:
```suggestion
                        assert "appsec" in products, "Product information is not accurately reported by telemetry on app-started event"
```

... and the try catch is useless


Though, if this test should always be true (no conditional presence), and you only want to test presence, it may be more easy to modify schemas. I can help you if needed